src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedMathUtils.java

```diff
@@ -136,4 +141,22 @@ public static double stdDev(final double ... values) {
         Utils.nonNull(values);
         return Math.sqrt(new Variance().evaluate(values));
     }
+
+    // given a list of options and a function for calculating their probabilities (these must sum to 1 over the whole list)
+    // randomly choose one option from the implied categorical distribution
+    public static <E> E randomSelect(final List<E> choices, final Function<E, Double> probabilityFunction) {
```

Comment:
That's a fair point about the optimization.  If you want to keep your implementation, though, I still think passing the rng would be a good idea.